pkg/collector/corechecks/cluster/kubernetes_apiserver.go

```diff
@@ -143,6 +146,21 @@ func (k *KubeASCheck) Run() error {
 		return err
 	}
 
+	if versionToken == "0" {
+		// API server cache expired or no recent events to process. Resetting the Resversion token.
+		newEvents, modifiedEvents, versionToken, err = asclient.LatestEvents("0")
+		if err != nil {
+			k.Warnf("Could not collect cached events from the api server: %s", err.Error())
+			return err
+		}
+		if k.latestEventToken == versionToken {
+			// No new events but protobuf error was caught. Will retry at next run.
+			return nil
+		}
+		// There was a protobuf error and new events were submitted. Processing them and updating the resVersion.
```

Comment:
there was or there was not?

Reply:
yeah there was, if versionToken is "0" it means it was forced. This would typically happen if we receive an error and thus we got stuck missing potential events. Now, we avoid getting stuck but we only continue the check if there are events to process.